skills_ml/algorithms/occupation_classifiers/train.py

```diff
@@ -0,0 +1,112 @@
+from gensim.models.doc2vec import Doc2Vec
+from gensim import __version__ as gensim_version
+from gensim import __name__ as gensim_name
+
+from skills_ml.datasets import job_postings
+from skills_ml.algorithms.corpus_creators.basic import Doc2VecGensimCorpusCreator
+
+from skills_utils.s3 import upload
+
+from datetime import datetime
+from itertools import chain
+from glob import glob
+
+import os
+import json
+import logging
+logging.basicConfig(format='%(asctime)s : %(levelname)s : %(message)s', level=logging.INFO)
+
+class RepresentationTrainer(object):
+    """A representation learning object using gensim doc2vec model.
+    Example:
+
+    from airflow.hooks import S3Hook
+    from skills_ml.algorithms.occupation_classifiers.train import RepresentationTrainer
+
+    s3_conn = S3Hook().get_conn()
+    trainer = RepresentationTrainer(s3_conn, ['2011Q1', '2011Q2'], 'open-skills-private/test_corpus')
+    trainer.train()
+    """
+    def __init__(self, s3_conn, quarters, jb_s3_path, model_s3_path='open-skills-private/model_cache'):
+        """Initialization
+
+        Attributes:
+            s3_conn (:obj: `boto.s3.connection.S3Connection`): the boto object to connect to S3.
+            quarters (:obj: `list` of (str)): quarters will be trained on
+            jb_s3_path (:str): job posting path on S3
+            model (:obj: `gensim.models.doc2vec.Doc2Vec`): gensim doc2vec model object
+            metadata (:dict): model metadata
+            training_time (:str): training time
+        """
+        self.s3_conn = s3_conn
+        self.quarters = quarters
+        self.jb_s3_path = jb_s3_path
+        self.model = None
+        self._metadata = None
+        self.training_time = datetime.today().isoformat()
+        self.model_s3_path = model_s3_path
+
+    def train(self, size=500, min_count=3, iter=8, window=6, workers=2, **kwargs):
+        """Train a doc2vec model, build a lookup table and model metadata. After training, they will be saved to S3.
+
+        Args:
+            kwargs: all arguments that gensim.models.doc2vec.Docvec will take.
+        """
+        generators = []
+        for quarter in self.quarters:
+            generators.append(job_postings(self.s3_conn, quarter, self.jb_s3_path))
```

Comment:
It would be helpful to be able to pass in the job posting generator *function*. Sometimes either `job_postings` or `job_postings_highmem` is more suitable depending on the environment, and it's nice to keep it configurable.